rosbag2_cpp/src/rosbag2_cpp/cache/message_cache_circular_buffer.cpp

```diff
@@ -25,25 +25,44 @@ namespace rosbag2_cpp
 namespace cache
 {
 
-MessageCacheCircularBuffer::MessageCacheCircularBuffer(size_t max_cache_size)
-: max_bytes_size_(max_cache_size)
+MessageCacheCircularBuffer::MessageCacheCircularBuffer(
+  size_t max_cache_size,
+  int64_t max_cache_duration_ns)
+: max_bytes_size_(max_cache_size), max_cache_duration_(max_cache_duration_ns)
 {
+  if (max_bytes_size_ == 0 && max_cache_duration_ == 0) {
+    ROSBAG2_CPP_LOG_ERROR_STREAM("Invalid arguments for the MessageCacheCircularBuffer. "
+                                 "Both max_bytes_size and max_cache_duration are zero.");
```

Comment:
if this is debug message, it should include some extra debug information that the following exception cannot throw as in the message? otherwise, it is the same information with the following exception, so maybe debugging message here is not really useful since user can see the exact same message via exception?